src/orange/orangeinp/ConvexHullFinder.hh

```diff
@@ -0,0 +1,281 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/orangeinp/ConvexHullFinder.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <algorithm>
+
+#include "corecel/Assert.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/Array.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/math/SoftEqual.hh"
+
+namespace celeritas
+{
+namespace orangeinp
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Find the convex hull of a sequence of 2D points.
+ *
+ * These points must be supplied in clockwise-order such that segments between
+ * adjacent points, including the last and first points, comprise a
+ * non-self-intersecting polygon. Exploiting this ordering, the Graham Scan
+ * algorithm \citet{graham-convexhull-1972,
+ * https://doi.org/10.1016/0020-0190(72)90045-2} finds the convex hull with
+ * O(N) time complexity.
+ */
+template<class T>
+class ConvexHullFinder
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using real_type = T;
+    using Real2 = celeritas::Array<real_type, 2>;
+    using VecReal2 = std::vector<Real2>;
+    using VecVecReal2 = std::vector<VecReal2>;
+    //!@}
+
+  public:
+    // Construct with vector of ordered points
+    explicit ConvexHullFinder(VecReal2 const& points);
```

Comment:
Perhaps also pass in a `Tolerance` object? Deletion of small concave regions (or near collinearity of points) may be affected by the user tolerance.